Common/Extensions/StreamExtensions.cs

```diff
@@ -380,6 +381,11 @@ public static DownloadAdapter WebClientDownload(System.Uri location, System.Net.
 
                         return d;
                     }
+                    catch (System.Exception ex)
+                    {
+                        d.Dispose();
+                        throw;
+                    }
                 }
```

Comment:
Missing a Finally here.

Reply:
This should be slightly different, try, catch, finally, if not using `using` :) Please revise.

Reply:
> Missing a Finally here.

Idea is to `Dispose` (not leak) `DownloadAdapter` only in case of exception thrown after `DownloadAdapter` created but before it is returned from emthod. So in case code between throws we do not leak resources. When no exceptions thrown we should return `DownloadAdapter` to use in caller.

Reply:
I understand now, I see... Thank you for your contributions!